docs/contributing/coding-style.md

```diff
@@ -508,6 +508,24 @@ func WithHosts(hosts ...string) Option {
 }
 ```
 
+If the functional option error is a critical error, we should wrap it with the `ErrCriticalOption` error and return it to the caller side to decide the next action.
+
+```go
+func WithConnectTimeout(dur string) Option {
+    return func(c *client) error {
+        if dur == "" {
+            return errors.ErrInvalidOption("connectTimeout", dur)
+        }
+        d, err := timeutil.Parse(dur)
+        if err != nil {
+            return errors.NewErrCriticalOption(err)
```

Comment:
should we wrap `ErrInvalidOption `??
I think we don't notice which option error.:thinking:

Reply:
should we change `NewErrCriticalOption(err error)` to `NewErrCriticalOption(opname string, val interface{}, err error)`

Reply:
@kevindiu 
```
package errors

var (
	// ErrInvalidOption returns invalid option error.
	ErrInvalidOption = func(name string, val interface{}) error {
		if val == nil {
			return Errorf("invalid option. name: %s, val: nil", name)
		}
		return Errorf("invalid option. name: %s, val: %#v", name, val)
	}

	// ErrInvalidOptionWithError --
	ErrInvalidOptionWithError = func(name string, val interface{}, err error) error {
		optErr := ErrInvalidOption(name, val)
		return Wrap(optErr, err.Error())
	}
)

// ErrCriticalOption represent the critical option error
type ErrCriticalOption struct {
	err error
}

func NewErrCriticalOption(name string, val interface{}) error {
	return &ErrCriticalOption{
		err: ErrInvalidOption(name, val),
	}
}

// NewErrCriticalOptionWithError --
func NewErrCriticalOptionWithError(name string, val interface{}, err error) error {
	return &ErrCriticalOption{
		err: ErrInvalidOptionWithError(name, val, err),
	}
}

func (e *ErrCriticalOption) Error() string {
	return Wrap(e.err, "invalid critical option").Error()
}

func IsCriticalOptionError(err error) bool {
	switch err.(type) {
	case *ErrCriticalOption:
		return true
	default:
		return false
	}
}

```

cc: @vankichi